docs/howtos/convert_pytorch_to_flax.rst

```diff
@@ -0,0 +1,351 @@
+  conv_bias = t_model.state_dict()['conv.bias'].detach().cpu().numpy()
+  fc_kernel = t_model.state_dict()['fc.weight'].detach().cpu().numpy()
+  fc_bias = t_model.state_dict()['fc.bias'].detach().cpu().numpy()
+
+  # [outC, inC, kH, kW] -> [kH, kW, inC, outC]
+  conv_kernel = jnp.transpose(conv_kernel, (2, 3, 1, 0))
+  
+  # [outC, inC] -> [inC, outC]
+  fc_kernel = jnp.transpose(fc_kernel, (1, 0))
+  
+  variables = {'params': {'conv': {'kernel': conv_kernel, 'bias': conv_bias},
+                          'fc': {'kernel': fc_kernel, 'bias': fc_bias}}}
+
+  key = random.PRNGKey(0)
+  x = random.normal(key, (1, 6, 6, 3))
+
+  j_out = j_model.apply(variables, x)
+  
+  # [N, H, W, C] -> [N, C, H, W]
+  t_x = torch.from_numpy(np.transpose(np.array(x), (0, 3, 1, 2)))
+  t_out = t_model(t_x)
+  t_out = t_out.detach().cpu().numpy()
+  
+  np.testing.assert_almost_equal(j_out, t_out, decimal=6)
+
+
+
+Batch Norm
+--------------------------------
+
+``torch.nn.BatchNorm2d`` uses ``0.1`` as the default value for the ``momentum`` parameter while
+|nn.BatchNorm|_ uses ``0.9``. However, this corresponds to the same computation, because PyTorch multiplies
+the estimated statistic with ``(1 − momentum)`` and the new observed value with ``momentum``,
+while Flax multiplies the estimated statistic with ``momentum`` and the new observed value with ``(1 − momentum)``.
+
+.. |nn.BatchNorm| replace:: ``nn.BatchNorm``
+.. _nn.BatchNorm: https://flax.readthedocs.io/en/latest/_autosummary/flax.linen.BatchNorm.html
+
+.. testcode::
+
+  t_bn = torch.nn.BatchNorm2d(num_features=3, momentum=0.1)
+  t_bn.eval()
+  
+  scale = t_bn.weight.detach().cpu().numpy()
+  bias = t_bn.bias.detach().cpu().numpy()
+  mean = t_bn.running_mean.detach().cpu().numpy()
+  var = t_bn.running_var.detach().cpu().numpy()
+  
+  variables = {'params': {'scale': scale, 'bias': bias},
+               'batch_stats': {'mean': mean, 'var': var}}
+  
+  key = random.PRNGKey(0)
+  x = random.normal(key, (1, 6, 6, 3))
+  
+  j_bn = nn.BatchNorm(momentum=0.9, use_running_average=True)
+  
+  j_out = j_bn.apply(variables, x)
+  
+  # [N, H, W, C] -> [N, C, H, W]
+  t_x = torch.from_numpy(np.transpose(np.array(x), (0, 3, 1, 2)))
+  t_out = t_bn(t_x)
+  # [N, C, H, W] -> [N, H, W, C]
+  t_out = np.transpose(t_out.detach().cpu().numpy(), (0, 2, 3, 1))
+  
+  np.testing.assert_almost_equal(j_out, t_out)  
+
+
+
+Average Pooling
+--------------------------------
+
+``torch.nn.AvgPool2d`` and |nn.avg_pool()|_ are compatible when using default parameters.
+However, ``torch.nn.AvgPool2d`` has a parameter ``count_include_pad``. When ``count_include_pad=False``,
+the zero-padding will not be considered for the average calculation. There does not exist a similar
+parameter for |nn.avg_pool()|_. However, we can easily implement a wrapper around the pooling
+operation.
+
+
+.. testcode::
+
+  def pool(inputs, init, reduce_fn, window_shape, strides, padding):
```

Comment:
In the interest of keeping this short and easy to read, let's instead export the function as `nn.pool()`